From 3a5a4be66b66d26c2e7eeee2e3112c17f1e64e18 Mon Sep 17 00:00:00 2001
From: Clayton Craft <clayton@craftyguy.net>
Date: Fri, 20 Feb 2026 20:48:38 -0800
Subject: [PATCH] pages/account: fall back to g_str_to_ascii() when iconv
 TRANSLIT is unavailable

ASCII//TRANSLIT isn't available on musl, and g_convert_with_fallback
seems to always return NULL when this is the case. This adds a fallback
when NULL is detected. This fixes an immediate and sudden crash when
inputing any character on the "Full Name" field on the account page.
---
 gnome-initial-setup/pages/account/um-utils.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gnome-initial-setup/pages/account/um-utils.c b/gnome-initial-setup/pages/account/um-utils.c
index e9941b39..b25fb024 100644
--- a/gnome-initial-setup/pages/account/um-utils.c
+++ b/gnome-initial-setup/pages/account/um-utils.c
@@ -194,6 +194,8 @@ generate_username_choices (const gchar  *name,
 
         ascii_name = g_convert_with_fallback (name, -1, "ASCII//TRANSLIT", "UTF-8",
                                               unicode_fallback, NULL, NULL, NULL);
+        if (ascii_name == NULL)
+                ascii_name = g_str_to_ascii (name, NULL);
 
         lc_name = g_ascii_strdown (ascii_name, -1);
 
-- 
2.53.0

